[TAU package] Enables the user to force either the new or legacy L0 profiler

jordialcaraz opened the pull request:
@spackbot fix style

jordialcaraz committed:
[TAU package] Enables the user to force either the new or legacy L0 profiler

jordialcaraz commented:
@spackbot fix style

jordialcaraz committed:
[@spackbot] updating style on behalf of jordialcaraz

jordialcaraz committed:
Merge branch 'develop' into develop

wspear reviewed (commented):
Is it possible to detect when new vs legacy is required based on the L0 dependency being used, making the variants unnecessary? If not, what about making 'new' the default and only providing an additional variant to specify legacy L0?  It'd be nice if the variant description could include a very succinct explanation of why you would want to use the legacy version.

jordialcaraz commented:
By default, it automatically detects what it should use.
The flags are in the case you want to use the old profiler, or when building the package using a machine which doesn't have a GPU. If there is no GPU, the check fails and the old profiler is used, so this is a workaround.

jordialcaraz committed:
Removes a variant

By default use the new L0 profiler.
+force_legacy_l0 variant disabled the new profiler and uses the legacy profiler

jordialcaraz commented:
@spackbot fix style

jordialcaraz committed:
[@spackbot] updating style on behalf of jordialcaraz

jordialcaraz committed:
Style fixes

jordialcaraz commented:
@spackbot fix style

jordialcaraz committed:
Merge branch 'develop' into develop

jordialcaraz commented:
The "when" is missing the ":", after 2.35. Cannot change it at this moment.

________________________________
From: spackbot-app[bot] ***@***.***>
Sent: Monday, November 3, 2025 8:35:03 PM
To: spack/spack-packages ***@***.***>
Cc: jordialcaraz ***@***.***>; Author ***@***.***>
Subject: Re: [spack/spack-packages] [TAU package] Enables the user to force either the new or legacy L0 profiler (PR #2271)

[https://avatars.githubusercontent.com/in/123749?s=20&v=4]spackbot-app[bot] left a comment (spack/spack-packages#2271)<https://github.com/spack/spack-packages/pull/2271#issuecomment-3482175124>

I was able to run spack style --fix for you!

spack style --fix

repos/spack_repo/builtin/packages/arrow/package.py
repos/spack_repo/builtin/packages/detray/package.py
repos/spack_repo/builtin/packages/glx/package.py
repos/spack_repo/builtin/packages/libxslt/package.py
repos/spack_repo/builtin/packages/tau/package.py
�[1;34m==> �[0mrunning flake8
�[1;34m==> �[0mrunning isort
�[1;34m==> �[0mrunning black
�[1;34m==> �[0mstyle checks passed


Keep in mind that I cannot fix your flake8 or mypy errors, so if you have any you'll need to fix them and update the pull request. If I was able to push to your branch, if you make further changes you will need to pull from your updated branch before pushing again.

I wasn't able to make any further changes, but please see the message above for remaining issues you can fix locally!

—
Reply to this email directly, view it on GitHub<https://github.com/spack/spack-packages/pull/2271#issuecomment-3482175124>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADTYFXY6FJG75NS5V3NWBJT326U6PAVCNFSM6AAAAACLABVHB2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTIOBSGE3TKMJSGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

jordialcaraz committed:
Version change

eugeneswalker commented:
@spackbot run pipeline due to runner system failure unrelated to this PR in prior run

jordialcaraz commented:
Just as a comment, due to the removed variant, now by default it forces the use of the new profiler, this can be overwritten with `+force-legacy-l0`

eugeneswalker commented:
The TAU builds are failing everywhere with this PR:

Here are some example jobs failed:
https://gitlab.spack.io/spack/spack-packages/-/pipelines/1295377

jordialcaraz committed:
Added spec version check for new variant

jordialcaraz commented:
@spackbot fix style

jordialcaraz committed:
[@spackbot] updating style on behalf of jordialcaraz

jordialcaraz committed:
Merge branch 'develop' into develop

eugeneswalker reviewed (dismissed):
Looks great to me! @wspear Any final thoughts? Should we merge now?

wspear commented:
@jordialcaraz correct me if I'm wrong, but it looks like:
```
        if spec.satisfies("@2.35:"):
            if "+force-legacy-l0" in spec:
                options.append("-force_legacy_l0")
            else:
                options.append("-force_new_l0")
```

Runs even if level zero isn't specified. Should the first if clause include  "+level_zero" in spec?  Otherwise we'll get -force_new_l0 in every configuration. Even if that flag does nothing when level_zero isn't specified that would be an unnecessary and easily avoided complication.

jordialcaraz commented:
Yes, you are right, let me move it.

jordialcaraz committed:
Only check if the new variant is set, if level_zero is also enabled

jordialcaraz commented:
@spackbot fix style
